Engine/Graphics/DecalBuilder.cpp

```diff
@@ -127,33 +137,33 @@ char DecalBuilder::ApplyDecals(int light_level, char LocationFlags, stru154 *a4,
     static_AE4F60.Normal.z = v16->face_plane.vNormal.z;
     static_AE4F60.dist = v16->face_plane.dist;
     if (!pIndoorCameraD3D->GetFacetOrientation(
-            v16->polygonType, &static_AE4F60.Normal, &static_AE4F60.field_10,
-            &static_AE4F60.field_1C))
+        v16->polygonType, &static_AE4F60.Normal, &static_AE4F60.field_10,
+        &static_AE4F60.field_1C))
         log->Warning("Error: Failed to get the facet orientation");
 
     if (this->uNumDecals > 0) {
         // a6b = this->std__vector_30B00C;
         for (int i = 0; i < this->uNumDecals; ++i) {
-            // v21 = &pBloodsplatContainer->std__vector_pBloodsplats[*a6b];
+            // v21 = &pBloodsplatContainer->pBloodsplats_to_apply[*a6b];
             int point_light_level =
                 _43F5C8_get_point_light_level_with_respect_to_lights(
                     light_level, uSectorID,
-                    bloodsplat_container->std__vector_pBloodsplats[this->std__vector_30B00C[i]].x,
-                    bloodsplat_container->std__vector_pBloodsplats[this->std__vector_30B00C[i]].y,
-                    bloodsplat_container->std__vector_pBloodsplats[this->std__vector_30B00C[i]].z);
+                    bloodsplat_container->pBloodsplats_to_apply[this->std__vector_30B00C[i]].x,
```

Comment:
can you extract this->std__vector_30B00C[i] to an int variable?
in fact the reason this construct is repeated (and occasional code is duplicated) was due to a poor decompiler inittially